summarisation.py

```diff
@@ -0,0 +1,33 @@
+from langchain.chat_models import ChatOpenAI
+from langchain.document_loaders import WebBaseLoader
+from langchain.chains.summarize import load_summarize_chain
+
+loader = WebBaseLoader("https://lilianweng.github.io/posts/2023-06-23-agent/")
+docs = loader.load()
+
+llm = ChatOpenAI(temperature=0, model_name="gpt-3.5-turbo-16k")
+chain = load_summarize_chain(llm, chain_type="stuff")
+
+chain.run(docs)
+
+from langchain.chains.llm import LLMChain
+from langchain.prompts import PromptTemplate
+from langchain.chains.combine_documents.stuff import StuffDocumentsChain
+
+# Define prompt
+prompt_template = """Write a concise summary of the following:
+"{text}"
+CONCISE SUMMARY:"""
+prompt = PromptTemplate.from_template(prompt_template)
+
+# Define LLM chain
+llm = ChatOpenAI(temperature=0, model_name="gpt-3.5-turbo-16k")
+llm_chain = LLMChain(llm=llm, prompt=prompt)
+
+# Define StuffDocumentsChain
+stuff_chain = StuffDocumentsChain(
+    llm_chain=llm_chain, document_variable_name="text"
+)
+
+docs = loader.load()
+print(stuff_chain.run(docs))
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section of the code is defining a prompt template, initializing another `ChatOpenAI` model, defining an `LLMChain` with the model and prompt, defining a `StuffDocumentsChain` with the `LLMChain`, loading documents, and running the `StuffDocumentsChain` on the loaded documents. Similar to the previous section, there's no error handling for potential issues. Adding error handling would improve the robustness of the script.

```diff
from langchain.chains.llm import LLMChain
from langchain.prompts import PromptTemplate
from langchain.chains.combine_documents.stuff import StuffDocumentsChain

# Define prompt
prompt_template = """Write a concise summary of the following:
"{text}"
CONCISE SUMMARY:"""
prompt = PromptTemplate.from_template(prompt_template)

# Define LLM chain
llm = ChatOpenAI(temperature=0, model_name="gpt-3.5-turbo-16k")
llm_chain = LLMChain(llm=llm, prompt=prompt)

# Define StuffDocumentsChain
stuff_chain = StuffDocumentsChain(
    llm_chain=llm_chain, document_variable_name="text"
)

+ try:
docs = loader.load()
print(stuff_chain.run(docs))
+ except Exception as e:
+     print(f"An error occurred: {e}")
+     return
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->